oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineImpl.java

```diff
@@ -292,7 +290,38 @@ public Result executeQuery(
             }
         }
     }
-    
+
+    @Override
+    public Result executeQuery(
+            String statement, String language, long limit, long offset,
+            Map<String, ? extends PropertyValue> bindings,
+            Map<String, String> mappings) throws ParseException {
+        return executeQuery(statement, language, Optional.of(limit), Optional.of(offset), bindings, mappings);
+
+    }
+
+    private long getOffset(List<Query> queries, Optional<Long> passedOffset) {
```

Comment:
(I didn't compile this sorry) but could it be static?

Reply:
Both methods made static.